crates/meroctl/src/cli/config.rs

```diff
@@ -0,0 +1,216 @@
+            }
+        } else {
+            eyre::bail!("You have to initialize the node first \nRun command node init -n <NAME>");
+        }
+
+        let mdns = self.mdns && !self.no_mdns;
+
+        let mut listen: Vec<Multiaddr> = vec![];
+
+        match swarm_listen {
+            Some(data) => listen.extend(data),
+            None => {
+                for host in self.swarm_host {
+                    let host = format!(
+                        "/{}/{}",
+                        match host {
+                            std::net::IpAddr::V4(_) => "ip4",
+                            std::net::IpAddr::V6(_) => "ip6",
+                        },
+                        host,
+                    );
+                    listen.push(format!("{}/tcp/{}", host, self.swarm_port).parse()?);
+                    listen.push(format!("{}/udp/{}/quic-v1", host, self.swarm_port).parse()?);
+                }
+            }
+        }
+
+        let mut boot_nodes = vec![];
+        if boot_network_provided {
+            if let Some(network) = self.boot_network {
+                match network {
+                    BootstrapNetwork::CalimeroDev => {
+                        boot_nodes.extend(BootstrapNodes::calimero_dev().list)
+                    }
+                    BootstrapNetwork::Ipfs => boot_nodes.extend(BootstrapNodes::ipfs().list),
+                }
+            }
+        } else {
+            boot_nodes = self.boot_nodes;
+        }
+
+        let config_new = ConfigFile {
+            identity,
+            store: StoreConfig {
+                path: "data".into(),
+            },
+            application: ApplicationConfig {
+                path: "apps".into(),
+            },
+            network: NetworkConfig {
+                swarm: SwarmConfig { listen },
+                bootstrap: BootstrapConfig {
+                    nodes: BootstrapNodes { list: boot_nodes },
+                },
+                discovery: DiscoveryConfig {
+                    mdns,
+                    rendezvous: Default::default(),
+                },
+                server: ServerConfig {
+                    listen: match server_listen {
+                        Some(data) => data,
+                        None => self
+                            .server_host
+                            .into_iter()
+                            .map(|host| {
+                                Multiaddr::from(host)
+                                    .with(multiaddr::Protocol::Tcp(self.server_port))
+                            })
+                            .collect(),
+                    },
+                    admin: Some(calimero_server::admin::service::AdminConfig { enabled: true }),
+                    jsonrpc: Some(calimero_server::jsonrpc::JsonRpcConfig { enabled: true }),
+                    websocket: Some(calimero_server::ws::WsConfig { enabled: true }),
+                },
+            },
+        };
+
+        config_new.save(&path)?;
```

Comment:
we should also be using `toml_edit` to preserve user-modifications like comments, whitespace in their config file

https://docs.rs/toml-edit